spec/models/user_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Supreeti') }
+
+  before { subject.save }
+
+  describe 'validation tests' do
+    it 'name should be present' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should be an int' do
+      subject.posts_counter = 'hey'
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should be greater than or equal to zero' do
+      subject.posts_counter = -1
+      expect(subject).to_not be_valid
```

Comment:
- [ ] Great job working on the tests but I think they are not passing because of the migrations so I think it would be great if you re-create them again and then run your tests ✨🙏🏻🔥

- I think you have conflicts in your migrations
- Posts
![image](https://github.com/supreeti/BlogApp-ROR/assets/58553711/fb397763-5fa6-461d-a713-f0f81b6943e0)
- Comments
![image](https://github.com/supreeti/BlogApp-ROR/assets/58553711/c9a0c021-51ed-4868-9315-e57435d2b6b2)
- Likes
![image](https://github.com/supreeti/BlogApp-ROR/assets/58553711/91a94c82-721b-4167-9791-93bebc6fad13)
 


![image](https://github.com/supreeti/BlogApp-ROR/assets/58553711/b8e61ba1-78d8-4fc4-b21a-21d0bca3f954)